core/src/main/java/org/infinispan/stream/StreamMarshalling.java

```diff
@@ -138,7 +169,7 @@ public Integer getId() {
       public void writeObject(ObjectOutput output, Object object) throws IOException {
          int number = objects.get(object.getClass(), -1);
          output.writeByte(number);
-         switch (number) {
+         switch (ExternalizerId.values()[number]) {
```

Comment:
Hrmm I can tweak the map to contain the enum instead.  Although it should be impossible to get a null here, note the should, haha.